### DISCUSSION TITLE
How to install on SD Card instead of emmc of Xavier NX

### DISCUSSION THREAD

**Discussion by TIS-Stefan** [opendatacam/opendatacam#626] [Q&A]
Hello

I try to install the docker container on my Xavier NX edge cam. The emmc has 15GB with 4.7GB free memory only, thus, the install script fails installing, because the disc is full. Therefore a 128 GB SD Card was added as /mnt/sdcard.

How do I get the install script to install on the SD Card? It still downloads to the emmc, even if starting the script from the SD Card. 

Thank you in advance!

Stefan

**Comment by vsaw** [comment]
Since the script is using docker to download and run the containers, OpenDataCam will always be downloaded to your Docker Data Root Folder, regardless where you execute the script. 

So what you will need to do is change docket data directory. The Docker docs tell you how https://docs.docker.com/config/daemon/#daemon-data-directory

**Comment by TIS-Stefan** [comment]
Hello

That was helpful. 
I changed the data-root in daemon.json, but on the wrong location. Then I missed a comma... The "panic: ... nil" error message of "docker info" was not that helpful in that situation. 

Now installed successfully!

Stefan

**Comment by TIS-Stefan** [comment]
Hello 

after getting it running, the Ubuntu on my Edge Cam became unusable. The terminal can not be opened, because permission denied on PTY. 
I do not know, what I did to make that, but must somehow be my fault. 
Then I learned, the container does not have a Gstreamer plugin. So i would like to remove it. But without a working terminal, I have no real chance to do that. Also I do not know, how to do that. Reinstalling the the complete Edge Cam is an option, which I would like to avoid. 

I am sorry, but I am a total beginner with Docker.

Stefan

**Reply by vsaw** [reply]
Not sure I fully understand what you are experiencing. Do you have log or screenshot data?

Generally speaking the docker container contains everything you need to run OpenDataCam, so no need to install GStreamer plugins.

In terms of shutting down ODC, if you somehow manage to open a terminal (or SSH into the machine externally) go to the folder where you ran the install script and run `docker-compose down`.

**Comment by TIS-Stefan** [comment]
Hello Vsaw,

thank you for answering. Since the Edge Cam Jetpack is unusable now, I will reflash the its Xavier NX  and try to run OpenData without docker container. (The Edge Cam is made by us, so this is no problem.)
I am convinced, it is a OpenData good software, which ran well on a NVidia Nano five years ago. Also I learned a lot.

Thank you very very much!

Stefan

**Reply by vsaw** [reply]
Sure, just let me know if anything else comes up! 👍

**Comment by TIS-Stefan** [comment]
Thank you!

Stefan
